crates/bevy_ui/src/accessibility.rs

```diff
@@ -0,0 +1,117 @@
+use bevy_a11y::{
+    accesskit::{kurbo::Rect, Node as AccessKitNode, Role},
+    AccessibilityNode,
+};
+use bevy_app::{App, CoreStage, Plugin};
+
+use bevy_ecs::{
+    prelude::Entity,
+    query::{Changed, Or, Without},
+    system::{Commands, Query},
+};
+use bevy_hierarchy::Children;
+
+use bevy_text::Text;
+use bevy_transform::prelude::GlobalTransform;
+use bevy_utils::default;
+
+use crate::{
+    prelude::{Button, Label},
+    Node, UiImage,
+};
+
+fn calc_name(texts: &Query<&Text>, children: &Children) -> Option<Box<str>> {
+    let mut name = None;
+    for child in children.iter() {
+        if let Ok(text) = texts.get(*child) {
+            let values = text
+                .sections
+                .iter()
+                .map(|v| v.value.to_string())
+                .collect::<Vec<String>>();
+            name = Some(values.join(" "));
+        }
+    }
+    name.map(|v| v.into_boxed_str())
+}
+
+fn calc_bounds(
+    mut query: Query<
+        (&mut AccessibilityNode, &Node, &GlobalTransform),
+        Or<(Changed<Node>, Changed<GlobalTransform>)>,
+    >,
+) {
+    for (mut accessible, node, transform) in &mut query {
+        let bounds = Rect::new(
+            transform.translation().x.into(),
+            transform.translation().y.into(),
+            (transform.translation().x + node.calculated_size.x).into(),
+            (transform.translation().y + node.calculated_size.y).into(),
+        );
+        accessible.bounds = Some(bounds);
+    }
+}
+
+fn button_changed(
+    mut commands: Commands,
+    query: Query<(Entity, &Children), Changed<Button>>,
+    texts: Query<&Text>,
+) {
+    for (entity, children) in &query {
+        let node = AccessKitNode {
+            role: Role::Button,
+            name: calc_name(&texts, children),
+            ..default()
+        };
+        commands
+            .entity(entity)
+            .insert(AccessibilityNode::from(node));
```

Comment:
Nit: using potentially overwriting inserts like this will be substantially slower than branching on an `Option<AccessibilityNode>`.